### PR TITLE
Add geothermal probe component

### PR DESCRIPTION
**Pull request by hsteinacker-siz**
Version bump to v0.8.4 with the following changes:
* Add geothermal probe as new component. It acts like a storage and there are many precalculated input parameters for a variety of probe field configurations. Both the simplified and the extended version were successfully validated.
* Add new szenario with geothermal probe and heat pump.
* Add geothermal collector files. This component is not finished yet and needs some rework.
* Add standalone julia code to create figures from profile data
* Add julia packages Roots and Plots

### CODE REVIEW

File: src/energy_systems/heat_sources/geothermal_probes.jl [hunk 1]
```diff
@@ -1,315 +1,862 @@
...
+    alpha::Float64:                 convective heat transfer coefficient in pipe in current time step
+    fluid_reynolds_number::Float64  the current Reynolds number of the fluid in the pipe
+"""
+function calculate_alpha_pipe(unit::GeothermalProbes)
+    # calculate mass flow in pipe
+    power_in_out_per_pipe = wh_to_watts(abs(unit.energy_in_out_per_probe_meter[unit.time_index])) * unit.probe_depth / unit.probe_type  # W/pipe
+    temperature_spread = unit.energy_in_out_per_probe_meter[unit.time_index] > 0 ? unit.loading_temperature_spread : unit.unloading_temperature_spread
+    mass_flow_per_pipe = power_in_out_per_pipe / (unit.fluid_specific_heat_capacity * temperature_spread)  # kg/s
+
+    use_dynamic_fluid_properties = false
```
Comment (eott-siz) [inline src/energy_systems/heat_sources/geothermal_probes.jl:651]: Is this supposed to be customizable? Or a feature that's not working yet?

File: src/energy_systems/heat_sources/geothermal_heat_collector_mesh_automatization.jl [hunk 2]
```diff
@@ -0,0 +1,209 @@
+# Mesh automatization
```
[the remark this answers is not in the document]
Comment (hsteinacker-siz) [reply src/energy_systems/heat_sources/geothermal_heat_collector_mesh_automatization.jl:1]: Yes, this file needs some revision and will be moved to initialize() of geothermal collector in a future release

File: src/energy_systems/heat_sources/geothermal_probes.jl [hunk 3]
```diff
@@ -1,315 +1,862 @@
...
+                                      "zoned_rectangle" => "zoned_rectangle_5m_v1.0.json",
+                                      "U_configurations" => "U_configurations_5m_v1.0.json",
+                                      "lopsided_U_configuration" => "LopU_configurations_5m_v1.0.json",
+                                      "C_configuration" => "C_configurations_5m_v1.0.json",
+                                      "L_configuration" => "L_configurations_5m_v1.0.json"
+                                      )
+
+    if !haskey(probe_field_configurations, unit.probe_field_geometry)
+        @error "The entered probe field configuration could not be detected in \"$(unit.uac)\". It has to be one of: $(join(["$key" for key in keys(probe_field_configurations)], ", "))."
+        exit()
```
[the remark this answers is not in the document]
Comment (hsteinacker-siz) [reply src/energy_systems/heat_sources/geothermal_probes.jl]: I have added a new exception "InputError" and redirected all exit() in Julia to this custom exception.

File: src/energy_systems/heat_sources/geothermal_probes.jl [hunk 4]
```diff
@@ -1,315 +1,862 @@
...
+                    ((1 + 2 * pi * unit.grout_heat_conductivity * beta) / (1 - 2 * pi * unit.grout_heat_conductivity * beta) +
+                    unit.radius_pipe_outer^2 / (4 * unit.distance_pipe_center^2) * (1 + sigma * 16 * unit.radius_borehole^4 * unit.distance_pipe_center^4 / ((unit.radius_borehole^4 - unit.distance_pipe_center^4)^2))))
+
+        borehole_thermal_resistance = R_1 / (2 * unit.probe_type)
+    elseif unit.model_type == "simplified"
+        # constant borehole thermal resistance from user input
+        borehole_thermal_resistance = unit.borehole_thermal_resistance
+    else
+        @error "Undefined model type $(unit.model_type) of unit $(unit.uac). Has to be one of: 'simplified', 'detailed'."
+        exit()
```
[the remark this answers is not in the document]
Comment (hsteinacker-siz) [reply src/energy_systems/heat_sources/geothermal_probes.jl]: I have moved the checks of input strings to the initialisation of the struct as long as we dont have a global input check.

File: src/energy_systems/heat_sources/geothermal_probes.jl (hunk 1 above)
Comment (hsteinacker-siz) [reply src/energy_systems/heat_sources/geothermal_probes.jl]: I have remove the flag and commented out the dynamic fluid parameters for potential further improvements of the model.

File: src/energy_systems/heat_sources/geothermal_probes.jl [hunk 5]
```diff
@@ -1,315 +1,862 @@
...
-This implementations acts as storage as is can produce and load energy.
-## ATTENTION: Geothermal probes are currently work in progress and not completed!!
-## ATTENTION: The physical implementation is party nonsense!!
+This implementations acts as storage as it can produce and load energy.
 """
+
+using JSON
+using Interpolations
+using Plots: plot, scatter, savefig
+import Plots
```
[the remark this answers is not in the document]
Comment (hsteinacker-siz) [reply src/energy_systems/heat_sources/geothermal_probes.jl:9]: I have streamlined the initialize!() of the geothermal probe for better readability. The plots are now separated and callable from ReSiE.jl using dump_auxiliary_outputs(). I have added global parameters to control the creation of plots of components. Every component can now have a plot_optional_figures() that will be called after initialize!() if activated globally.

**Review by eott-siz — APPROVED** [review]
Comments addressed as discussed. Looks good now!

**Pull request merged**